blocks/bricks/recurrent.py

```diff
@@ -630,3 +631,241 @@ def apply(self, *args, **kwargs):
+        called "states". To be more precise, this is the name of the
+        element in the outputs of the apply method of each transition
+        (layer) that is used, via fork, as the sequences (input) of the
+        next layer. The same element should also appear in the states
+        parameter of the apply method.
+    fast : bool
+        Use the fast, but also memory consuming, implementation of this
+        code. By default true.
+    skip_connections : bool
+        By default False. When true, the input (sequences) is spread to
+        all layer and not just the bottom layer. Each vector in the
+        sequences to the first layer is passed through a linear
+        transformation, without bias, to each of the sequences of the
+        other layers and is added to the
+        usually input comfing from a fork from the states of the previous
+        layer.
+        The code asserts that all layers have the same names in their
+        sequences and in the same order.
+
+    Notes
+    -----
+    See :class:`.BaseRecurrent` for more initialization parameters.
+
+    """
+    def __init__(self, transitions, fork_prototype=None, states_name="states",
+                 fast=True, skip_connections=False, **kwargs):
+        super(RecurrentStack, self).__init__(**kwargs)
+
+        self.states_name = states_name
+
+        for d, transition in enumerate(transitions):
+            transition.name += '_' + str(d)
+        self.transitions = transitions
+
+        if fork_prototype is None:
+            # By default use Linear (with bias) as a default fork.
+            # This overrides the bad default inside Fork which is without bias.
+            # Yes I know use_bias=True us the default, but it is a big deal so
+            # I write it down explicitly.
+            fork_prototype = Linear(use_bias=True)
+        depth = len(transitions)
+        self.forks = [Fork(self.normal_inputs(d), name='fork_' + str(d),
+                           prototype=fork_prototype)
+                      for d in range(1, depth)]
+
+        self.skip_connections = []
+        if skip_connections:
+            input_names = self.normal_inputs(0)
+            input_dims = self.transitions[0].get_dims(input_names)
+            for d in range(1, depth):
+                for name_d, name_0 in zip(self.normal_inputs(d), input_names):
+                    assert name_d == name_0
+                self.skip_connections.append(Parallel(
+                    input_names, input_dims,
+                    self.transitions[d].get_dims(input_names),
+                    child_prefix='skip_' + str(d)))
+
+        self.children = self.transitions + self.forks + self.skip_connections
+
+        # Programmatically set the apply method
+        self.apply = self.fast_apply if fast else self.low_memory_apply
+        setattr(self.apply, 'sequences', transitions[0].apply.sequences)
+        for t in ["states", "contexts", "outputs"]:
```

Comment:
This piece of code needs refactoring. We try not to use abbreviations (except very common ones) and one character identifies in Blocks. You can use `level` instead of `d`, `name` instead of `s`, `property_` instead of `t`, etc.